lib/bounces.js

```diff
@@ -49,6 +49,44 @@ module.exports = function (log, error) {
       }
     }
 
+    function getHeaderValue(headerName, message){
+      var value = ''
+      if (message.mail && message.mail.headers) {
+        message.mail.headers.some(function (header) {
+          if (header.name === headerName) {
+            value = header.value
+            return true
+          }
+
+          return false
+        })
+      }
+
+      return value
+    }
+
+    function logBounceEvent(message) {
+      var verifyCode = getHeaderValue('X-Verify-Code', message)
+      var uid = getHeaderValue('X-Uid', message)
+
+      if (uid && verifyCode) {
+        var fakeRequestObject = {
+          auth: {
+            credentials: {
+              uid: uid,
+              id: verifyCode
+            }
+          },
+          headers: message.headers
+        }
+
+        return log.activityEvent('account.confirmation.bounced', fakeRequestObject, {
```

Comment:
FWIW, while diving through signin unblock, I found that nothing was calling `flowEvent` directly, it was only used from inside `log.activityEvent`. There may be a desire to improve that, but it'd seem like deviating from the norm in this case, no?